### PR TITLE
optimize two-stage merge

### PR DESCRIPTION
**Pull request by murphyatwork**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] others

## Changes
1. Add an interface `Column::append_permtuation`, to append data from multiple chunks
2. Change TopNNode's merge procedure to two stage: compare, and then merge data. The merge could utilize the `append_permutation` interface, which could amortize overhead.

### CODE REVIEW

File: be/src/column/column.h [hunk 1]
```diff
@@ -150,6 +150,9 @@ class Column {
         return append_selective(src, indexes.data(), 0, indexes.size());
     }
 
+    // Append rows from permutation
+    virtual void append_permutation(const Columns& columns, const Permutation& perm) = 0;
```
Comment (kangkaisen) [inline be/src/column/column.h:154]: Permutation is tight coupling with sort operator, so I think would better use visitor here.

**Comment by kangkaisen** [comment]
What'a the improvement effect of this PR?

File: be/src/column/column.h (hunk 1 above)
Comment (mofeiatwork) [reply be/src/column/column.h]: done

**Comment by mofeiatwork** [comment]
> What'a the improvement effect of this PR?

This PR depends on other PR to show the improvement, cause bottleneck is shifted to here if another bottleneck is eliminated.

**Review by kangkaisen — APPROVED** [review]

**Review by Pslydhh — APPROVED** [review]

File: be/src/exec/vectorized/sorting/sort_permute.cpp [hunk 2]
```diff
@@ -34,4 +50,159 @@ bool TieIterator::next() {
...
+        }
+
+        uint32_t orig_size = dst->size();
+        Columns null_columns, data_columns;
+        for (auto& col : _columns) {
+            const NullableColumn* src_column = down_cast<const NullableColumn*>(col.get());
+            null_columns.push_back(src_column->null_column());
+            data_columns.push_back(src_column->data_column());
+        }
+        if (_columns[0]->is_nullable()) {
```
Comment (trueeyu) [inline be/src/exec/vectorized/sorting/sort_permute.cpp:71]: why only check _columns[0] ?

**Review by trueeyu — APPROVED** [review]

**Pull request merged**